subprojects/gdk_rust/gdk_common/src/be/transaction.rs

```diff
@@ -0,0 +1,716 @@
+                for _ in 0..more_outputs {
+                    let new_out = elements::TxOut {
+                        asset: confidential::Asset::Confidential(0u8, [0u8; 32]),
+                        value: confidential::Value::Confidential(0u8, [0u8; 32]),
+                        nonce: confidential::Nonce::Confidential(0u8, [0u8; 32]),
+                        ..Default::default()
+                    };
+                    tx.output.push(new_out);
+                }
+                let sur_size = asset_surjectionproof_size(std::cmp::max(1, tx.input.len()));
+                for output in tx.output.iter_mut() {
+                    output.witness = TxOutWitness {
+                        surjection_proof: vec![0u8; sur_size],
+                        rangeproof: vec![0u8; 4174],
+                    };
+                    output.script_pubkey = vec![0u8; 21].into();
+                }
+
+                tx.output.push(elements::TxOut::default()); // mockup for the explicit fee output
+                let vbytes = tx.get_weight() as f64 / 4.0;
+                let fee_val = (vbytes * fee_rate * 1.03) as u64; // increasing estimated fee by 3% to stay over relay fee, TODO improve fee estimation and lower this
+                info!(
+                    "DUMMYTX inputs:{} outputs:{} num_changes:{} vbytes:{} sur_size:{} fee_val:{}",
+                    tx.input.len(),
+                    tx.output.len(),
+                    more_outputs,
+                    vbytes,
+                    sur_size,
+                    fee_val
+                );
+                fee_val
+            }
+        }
+    }
+
+    pub fn estimated_changes(&self, send_all: bool, wallet_data: &WalletData) -> u8 {
+        match self {
+            Self::Bitcoin(_) => 1u8 - send_all as u8,
+            Self::Elements(tx) => {
+                let mut different_assets = HashSet::new();
+                for input in tx.input.iter() {
+                    let asset_hex = wallet_data
+                        .all_txs
+                        .get_previous_output_asset_hex(
+                            input.previous_output,
+                            &wallet_data.all_unblinded,
+                        )
+                        .unwrap();
+                    different_assets.insert(asset_hex.clone());
+                }
+                if different_assets.is_empty() {
+                    0
+                } else {
+                    different_assets.len() as u8 - send_all as u8
+                }
+            }
+        }
+    }
+
+    /// return a Vector with the amount needed for this transaction to be valid
+    /// for bitcoin it contains max 1 element eg ("btc", 100)
+    /// for elements could contain more than 1 element, 1 for each asset, with the policy asset last
+    pub fn needs(
+        &self,
+        fee_rate: f64,
+        no_change: bool,
+        policy_asset: Option<String>,
+        wallet_data: &WalletData,
+    ) -> Vec<AssetValue> {
```

Comment:
probably BTreeMap since caller needs to pop, but probably yes since it would allow removing AssetValue type.
Adding a TODO comment in 15a94d1a83988cc939b4d09f37753ff209335b68